src/components/navigation/WalletLoaderHeader.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import * as React from 'react'
+import { Image } from 'react-native'
+import { AnimatedCircularProgress } from 'react-native-circular-progress'
+import { connect } from 'react-redux'
+
+import { getWalletLoadingPercent } from '../../modules/UI/selectors.js'
+import type { RootState } from '../../types/reduxTypes.js'
+import { type Theme, type ThemeProps, cacheStyles, withTheme } from '../services/ThemeContext.js'
+
+type StateProps = {
+  icon: string | void,
+  progress: number
+}
+
+type Props = StateProps & ThemeProps
+
+type State = {
+  isDone: boolean
+}
+
+export class WalletLoaderHeaderComponent extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = { isDone: false }
+  }
+
+  componentDidUpdate() {
+    if (this.props.progress === 100) {
+      setTimeout(() => {
+        this.setState({ isDone: true })
+      }, 5000)
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.progress === 100) {
+      this.setState({ isDone: true })
+    }
+  }
+
+  render() {
+    const { isDone } = this.state
+    const { icon, progress, theme } = this.props
+    const styles = getStyles(this.props.theme)
+    return (
+      <AnimatedCircularProgress
+        size={theme.rem(2.25)}
+        width={theme.rem(3 / 16)}
+        fill={progress}
+        tintColor={isDone ? theme.walletLoaderFillDone : theme.walletLoaderFill}
+        backgroundColor={theme.walletLoaderBackground}
+        rotation={0}
+      >
+        {() => <Image style={styles.icon} source={{ uri: icon }} />}
+      </AnimatedCircularProgress>
+    )
+  }
+}
+
+export const WalletLoaderHeader = connect((state: RootState): StateProps => {
```

Comment:
Change the name of component to WalletProgressIcon. This should be a reasonable component that we can use on the wallet rows

Reply:
Fixed